enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisteredRewriter.scala

```diff
@@ -39,6 +39,25 @@ class RegisteredRewriter(tokenContext: TokenContext) {
     val newPipelineInfo = mutable.HashMap[LogicalPlan, PipelineInformation]()
     var rewrites = Map[LogicalPlan, LogicalPlan]()
     val rewritePlanWithRegisters = topDown(Rewriter.lift {
+      /*
+      Projection means executing expressions and writing the result to a row. Since any expression of Variable-type
+      would just write to the row the data that is already in it, we can just skip them
+       */
+      case oldPlan@Projection(_, expressions) =>
+        val information = pipelineInformation(oldPlan)
+        val rewriter = rewriteCreator(information, oldPlan)
+
+        val newExpressions = expressions collect {
+          case (column, expression) if !expression.isInstanceOf[Variable] => column -> expression.endoRewrite(rewriter)
+        }
+
+        val newPlan = oldPlan.copy(expressions = newExpressions)(oldPlan.solved)
```

Comment:
Looking down at the ProjectionRegisteredPipe I see the parameter `introducedExpression`, and modification of the incoming context instead of creation of a new one, so this makes me begin to think the above code is correct in the sense that it only needs to know about the modified columns, which are the expressions to be run during projection. I am wondering however about projections that remove columns, but that is presumably already handled in the pipeline planning, which defines all columns that should be known about at the time.